packages/firebase_vertexai/firebase_vertexai/lib/src/live_api.dart

```diff
@@ -0,0 +1,302 @@
+///
+/// This class serves as a base for different types of server messages,
+/// such as content updates, tool calls, and tool call cancellations.
+/// Subclasses should implement specific message types.
+abstract class LiveServerMessage {}
+
+/// A message indicating that the live server setup is complete.
+///
+/// This message signals that the initial connection and setup process
+/// with the live server has finished successfully.
+class LiveServerSetupComplete implements LiveServerMessage {}
+
+/// Represents content generated by the model in a live stream.
+class LiveServerContent implements LiveServerMessage {
+  /// Creates a [LiveServerContent] instance.
+  ///
+  /// [modelTurn] (optional): The content generated by the model.
+  /// [turnComplete] (optional): Indicates if the turn is complete.
+  /// [interrupted] (optional): Indicates if the generation was interrupted.
+  LiveServerContent({this.modelTurn, this.turnComplete, this.interrupted});
+
+  /// The content generated by the model.
+  final Content? modelTurn;
+
+  /// Indicates if the turn is complete.
+  final bool? turnComplete;
+
+  /// Indicates if the generation was interrupted.
+  final bool? interrupted;
+}
+
+/// Represents a tool call in a live stream.
+class LiveServerToolCall implements LiveServerMessage {
+  /// Creates a [LiveServerToolCall] instance.
+  ///
+  /// [functionCalls] (optional): The list of function calls.
+  LiveServerToolCall({this.functionCalls});
+
+  /// The list of function calls.
+  final List<FunctionCall>? functionCalls;
+}
+
+/// Represents a tool call cancellation in a live stream.
+class LiveServerToolCallCancellation implements LiveServerMessage {
+  /// Creates a [LiveServerToolCallCancellation] instance.
+  ///
+  /// [functionIds] (optional): The list of function IDs to cancel.
+  LiveServerToolCallCancellation({this.functionIds});
+
+  /// The list of function IDs to cancel.
+  final List<String>? functionIds;
+}
+
+/// Represents realtime input from the client in a live stream.
+class LiveClientRealtimeInput {
+  /// Creates a [LiveClientRealtimeInput] instance.
+  ///
+  /// [mediaChunks] (optional): The list of media chunks.
+  LiveClientRealtimeInput({this.mediaChunks});
+
+  /// The list of media chunks.
+  final List<InlineDataPart>? mediaChunks;
+
+  // ignore: public_member_api_docs
+  Map<String, dynamic> toJson() => {
+        'realtime_input': {
+          'media_chunks':
+              mediaChunks?.map((e) => e.toMediaChunkJson()).toList(),
+        },
+      };
+}
+
+/// Represents content from the client in a live stream.
+class LiveClientContent {
+  /// Creates a [LiveClientContent] instance.
+  ///
+  /// [turns] (optional): The list of content turns from the client.
+  /// [turnComplete] (optional): Indicates if the turn is complete.
+  LiveClientContent({this.turns, this.turnComplete});
+
+  /// The list of content turns from the client.
+  final List<Content>? turns;
+
+  /// Indicates if the turn is complete.
```

Comment:
[nit]  https://dart.dev/effective-dart/documentation#prefer-starting-a-boolean-variable-or-property-comment-with-whether-followed-by-a-noun-or-gerund-phrase

```dart
/// Whether the turn is complete.
```

Though also consider removing this if it can't be expanded.

Reply:
Add more details.